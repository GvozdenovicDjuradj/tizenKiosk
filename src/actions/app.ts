import { Action, APP, APPLICATION_START } from "./types"
import { AppScreens } from "../interfaces"

export const applicationStart = (): Action => ({ type: APPLICATION_START })

export const initApp =  (): Action => ({type: APP.START.REQUEST})

export const stopApp = (): Action => ({ type: APP.SHUTDOWN.REQUEST })

export const changeInitialPage = (payload: AppScreens): Action => ({
  type: APP.CHANGE_INITIAL_PAGE.REQUEST,
  payload,
})

export const getPermissionReadPhoneState = () => ({
  type: APP.PHONE_STATE_PERMISSION.REQUEST
})

export const deactivateApp = (): Action => ({ type: APP.RESET.REQUEST })
