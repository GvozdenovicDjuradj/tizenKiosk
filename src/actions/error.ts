import { AnyAction } from "redux"
import { ERROR_SHOW, ERROR_RESET } from "./types"

export interface ShowError extends AnyAction {
  type: ERROR_SHOW;
  payload: string;
}

export interface ResetError extends AnyAction {
  type: ERROR_RESET
}

export type ErrorAction = ResetError | ShowError

export const showError = (payload: string): ShowError => ({
  type: ERROR_SHOW,
  payload
})

export const resetError = (): ResetError => ({ type: ERROR_RESET })
