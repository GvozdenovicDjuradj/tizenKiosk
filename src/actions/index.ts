export * from "./app"
export * from "./error"
export * from "./kiosk"
export * from "./kioskSettings"
export * from "./printing"
export * from "./navigationActions"
export * from "./screenSaver"
