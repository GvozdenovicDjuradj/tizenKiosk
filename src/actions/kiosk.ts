import { Action, AnyAction } from "redux"
import { APP, KIOSK, PRIVACY_POLICY_AGREE, PRIVACY_POLICY_DISAGREE } from "./types"
import {
  AddCustomerToQueueReset,
  AddCustomerToQueueSuccess,
  AddPrivateCustomerToQueueRequest,
  AddPrivateCustomerToQueueSuccess,
  AnyProduct,
  CustomerInQueue,
  DatePayload,
  ErrorPayload,
  FieldUpdatePayload,
  GetKioskDataSuccess,
  GetKioskSettingsSuccess,
  KioskSettings,
  Language,
  LanguagePayload,
  LoadPrivacyPolicySuccess,
  PrivacyPolicy,
  Product,
  ProductPayload,
  QuestionAnswerPayload,
  RegisterKioskFieldUpdate,
  RegisterKioskSuccess,
  StringPayload,
  NumberPayload
} from "../interfaces"
import { GET_KIOSK_ID_AND_DOMAIN_FROM_URL, SET_KIOSK_ID, SET_KIOSK_SERIAL, SET_KIOSK_URL } from "../constants/kiosk"

export const registerKiosk = (showPrint: boolean): AnyAction =>
  ({ type: KIOSK.REGISTER.REQUEST, hasPrinter: showPrint })

export const registerKioskFieldUpdate = (payload: FieldUpdatePayload): RegisterKioskFieldUpdate => ({
  payload,
  type: KIOSK.FIELD_UPDATE.REQUEST,
})

export const loadKioskSettings = (): Action => ({ type: KIOSK.SETTINGS.REQUEST })

export const loadScreenSaverData = (): Action => ({ type: KIOSK.LOAD_SCREEN_SAVER_DATA.REQUEST })

export const loadKioskQueueData = (): Action => ({ type: KIOSK.QUEUE_DATA.REQUEST })

interface KioskSettingsSuccessAction {
  type: string,
  payload: KioskSettings
}

export const kioskSettingsSuccess = (kioskSettings: KioskSettings): KioskSettingsSuccessAction => ({
  payload: kioskSettings,
  type: KIOSK.SETTINGS.SUCCESS
})

export const changeLanguage = (payload: Language): LanguagePayload => ({
  payload,
  type: KIOSK.SET_LANGUAGE.REQUEST,
})

export const setVenueCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.SET_COUNTRY.REQUEST
})

export const setGroupSize = (payload: number): NumberPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_GROUP_SIZE.REQUEST,
})

export const setCustomerName = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_NAME.REQUEST,
})

export const setCustomerEmail = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_EMAIL.REQUEST,
})

export const setCustomerCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_COUNTRY.REQUEST,
})

export const setCallingCode = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_CALLING_CODE.REQUEST,
})

export const setCustomerMobile = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_MOBILE.REQUEST,
})

export const setCustomerOrder = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_ORDER.REQUEST,
})

export const setNotes = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CUSTOMER.SET_NOTES.REQUEST,
})

export const setProduct = (payload: Product | AnyProduct): ProductPayload => ({
  payload,
  type: KIOSK.SET_PRODUCT.REQUEST,
})

export const setSubProduct = (payload: AnyProduct): ProductPayload => ({
  payload,
  type: KIOSK.SET_SUBPRODUCT.REQUEST,
})

export const addCustomerToQueue = (): Action => ({
  type: KIOSK.CUSTOMER.ADD_TO_QUEUE.REQUEST,
})

export const setAddCustomerToQueueError = (payload: string): StringPayload => ({
  type: KIOSK.CUSTOMER.ADD_TO_QUEUE.FAILURE,
  payload
})

export const setCheckInEmail = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_EMAIL.REQUEST,
  payload,
})

export const setCheckInMobile = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_MOBILE.REQUEST,
  payload,
})

export const setCheckInCountry = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CHECK_IN.SET_COUNTRY,
})

export const setCheckInCallingCode = (payload: string): StringPayload => ({
  payload,
  type: KIOSK.CHECK_IN.SET_CALLING_CODE,
})

export const setCheckInOrder = (payload: string): StringPayload => ({
  type: KIOSK.CHECK_IN.SET_ORDER.REQUEST,
  payload,
})

export const loadPrivacyPolicyDataSuccess = (payload: PrivacyPolicy, backgroundTask: boolean)
  : LoadPrivacyPolicySuccess => ({
    type: KIOSK.LOAD_PRIVACY_POLICY_DATA.SUCCESS,
    payload,
    backgroundTask
  })

export const addPrivateCustomerToQueueRequest = (queueId?: number, productId?: number)
  : AddPrivateCustomerToQueueRequest => ({
    type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.REQUEST,
    productId,
    queueId,
})

export const addPrivateCustomerToQueueError = (payload: ErrorPayload) => ({
  type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.FAILURE,
  payload
})

export const addPrivateCustomerToQueueSuccess = (customerInQueue: CustomerInQueue)
  : AddPrivateCustomerToQueueSuccess => ({
    type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.SUCCESS,
    payload: {
      customerInQueue
    }
  })

export const addCustomerToQueueSuccess = (customerInQueue: CustomerInQueue)
  : AddCustomerToQueueSuccess => ({
    type: KIOSK.CUSTOMER.ADD_TO_QUEUE.SUCCESS,
    payload: {
      customerInQueue
    }
  })

export const secretTap = () => ({
  type: APP.SECRET_TAP.REQUEST
})

export const addCustomerToQueueReset = (backgroundTask: boolean = false)
  : AddCustomerToQueueReset => ({
    type: KIOSK.CUSTOMER.ADD_TO_QUEUE.RESET,
    backgroundTask
  })

export const checkIn = (): Action => ({ type: KIOSK.CHECK_IN.REQUEST })

export const goToInitialScreen = (): Action => ({ type: APP.GO_INITIAL_SCREEN.REQUEST })

export const loadPrivacyPolicyDataRequest = (): Action => ({ type: KIOSK.LOAD_PRIVACY_POLICY_DATA.REQUEST })

export const agreePrivacyPolicy = (): Action => ({ type: PRIVACY_POLICY_AGREE })

export const disagreePrivacyPolicy = (): Action => ({ type: PRIVACY_POLICY_DISAGREE })

export const addPrivateCustomerToQueueFail = (): Action => ({ type: KIOSK.ADD_PRIVATE_CUSTOMER_TO_QUEUE.FAILURE })

export interface GetKioskIdAndDomainFromUrlAction {
    type: GET_KIOSK_ID_AND_DOMAIN_FROM_URL
}

export const getKioskIdAndDomainFromUrl = (): GetKioskIdAndDomainFromUrlAction => ({
    type: GET_KIOSK_ID_AND_DOMAIN_FROM_URL
})

export interface SetKioskIdAction {
    type: SET_KIOSK_ID,
    payload: {
        id: string;
    }
}

export const setKioskId = (id: string): SetKioskIdAction => ({
    type: SET_KIOSK_ID,
    payload: {
        id
    }
})

export interface SetKioskSerialAction extends Action {
    type: SET_KIOSK_SERIAL,
    payload: {
        serial: string;
    }
}

export const setKioskSerial = (serial: string): SetKioskSerialAction => ({
    type: SET_KIOSK_SERIAL,
    payload: {
        serial
    }
})

export interface SetKioskUrlAction extends Action {
  type: SET_KIOSK_URL,
  payload: {
    url: string;
  }
}

export const setKioskUrl = (url: string): SetKioskUrlAction => ({
  type: SET_KIOSK_URL,
  payload: {
    url
  }
})

export type Actions =
  | Action
  | DatePayload
  | GetKioskDataSuccess
  | GetKioskSettingsSuccess
  | LanguagePayload
  | ProductPayload
  | QuestionAnswerPayload
  | RegisterKioskFieldUpdate
  | RegisterKioskSuccess
  | StringPayload
  | SetKioskUrlAction
  | SetKioskSerialAction
  | SetKioskIdAction
