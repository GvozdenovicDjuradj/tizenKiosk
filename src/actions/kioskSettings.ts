import { Action } from "redux"
import { KIOSK_SETTINGS_FORM } from "./types"
import {
  FieldUpdatePayload,
  KioskSettingsFieldUpdate
} from "../interfaces"

export const updateSettingsField = (payload: FieldUpdatePayload): KioskSettingsFieldUpdate => ({
  type: KIOSK_SETTINGS_FORM.FIELD_UPDATE.REQUEST,
  payload
})

export const checkNetwork = (): Action => ({
  type: KIOSK_SETTINGS_FORM.CHECK_NETWORK.REQUEST
})

export const validateKioskSettings = (): Action => ({
  type: KIOSK_SETTINGS_FORM.VALIDATE.REQUEST
})

export const saveKioskSettingsChanges = (): Action => ({
  type: KIOSK_SETTINGS_FORM.SUBMIT.REQUEST
})
