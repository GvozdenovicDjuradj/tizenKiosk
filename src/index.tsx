import "babel-polyfill"
import { AppRegistry } from "react-native"
import App from "./App"
import app from "../app.json"

AppRegistry.registerComponent(app.name, () => App)

if (window && typeof window === "object" && window.document) {
  AppRegistry.runApplication(app.name, { rootTag: document.getElementById("app") })
}
