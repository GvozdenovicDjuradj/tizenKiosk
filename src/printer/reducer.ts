import { List } from "immutable"
import { Printer } from "@dnlowman/react-native-star-prnt"
import {
    APPEND_PRINTER_DEBUG_OUTPUT,
    CLEAR_PRINTER_DEBUG_OUTPUT,
    DISPLAY_PRINTER_NOTIFICATION,
    FIND_CONNECTED_PRINTERS,
    FIND_CONNECTED_PRINTERS_FULFILLED,
    FIND_CONNECTED_PRINTERS_REJECTED,
    HIDE_PRINTER_NOTIFICATION,
    SELECT_PRINTER,
    SET_DEBUG_OUTPUT,
    SET_PRINTER_ENABLED,
    TICKET_PRINTED,
    TOGGLE_DEBUG_OUTPUT,
    UPDATE_TICKET
} from "./constants"
import { PrinterActions } from "./actionCreators"
import { Reducer } from "redux"
import { APP } from "../actions/types"
import { PRINT_TICKET } from "../constants/printing"

const { freeze } = Object

export interface PrinterState {
    isFindingPrinters: boolean;
    connectedPrinters: List<Printer>;
    selectedPrinter?: Printer;
    logoUrl: string;
    ticketMessage: string;
    showAtTopValue: string;
    localCurrentTime: string;
    localCurrentDate: string;
    logoFooterUrl: string;
    isPendingPrint: boolean;
    isPrinterEnabled: boolean;
    debugOutput: string[];
    isDebugOutputVisible: boolean;
    notificationTitle: string;
    notificationMessage: string;
    isNotificationVisible: boolean;
    isCurrentPrintATest: boolean;
}

export const initialState: Readonly<PrinterState> = freeze({
    isFindingPrinters: false,
    connectedPrinters: List<Printer>(),
    selectedPrinter: undefined,
    logoUrl: "",
    ticketMessage: "",
    showAtTopValue: "",
    localCurrentTime: "",
    localCurrentDate: "",
    logoFooterUrl: "",
    isPendingPrint: false,
    isPrinterEnabled: false,
    debugOutput: [],
    isDebugOutputVisible: false,
    notificationTitle: "",
    notificationMessage: "",
    isNotificationVisible: false,
    isCurrentPrintATest: false,
})

const reducer: Reducer<PrinterState, PrinterActions> =
    (state: PrinterState = initialState, action: PrinterActions): Readonly<PrinterState> => {
        switch (action.type) {
            case PRINT_TICKET:
                return {
                    ...state,
                    isCurrentPrintATest: action.payload.isTest
                }

            case APP.SECRET_TAP.SUCCESS:
                return {
                    ...state,
                    isPendingPrint: false
                }

            case FIND_CONNECTED_PRINTERS:
                return {
                    ...state,
                    connectedPrinters: List.of<Printer>(),
                    isFindingPrinters: true,
                    selectedPrinter: undefined,
                }

            case FIND_CONNECTED_PRINTERS_FULFILLED:
                return {
                    ...state,
                    connectedPrinters: action.payload.printers,
                    isFindingPrinters: false,
                }

            case FIND_CONNECTED_PRINTERS_REJECTED:
                return {
                    ...state,
                    isFindingPrinters: false
                }

            case SELECT_PRINTER:
                return {
                    ...state,
                    selectedPrinter: action.payload.printer,
                }

            case UPDATE_TICKET:
                return {
                    ...state,
                    logoUrl: action.payload.logoUrl,
                    ticketMessage: action.payload.ticketMessage,
                    showAtTopValue: action.payload.showAtTopValue,
                    localCurrentTime: action.payload.localCurrentTime,
                    localCurrentDate: action.payload.localCurrentDate,
                    logoFooterUrl: action.payload.logoFooterUrl,
                    isPendingPrint: true
                }

            case TICKET_PRINTED:
                return {
                    ...state,
                    isPendingPrint: false,
                    isNotificationVisible: false,
                    notificationTitle: "",
                    notificationMessage: "",
                }

            case SET_PRINTER_ENABLED:
                return {
                    ...state,
                    isPrinterEnabled: action.payload.enabled,
                    selectedPrinter: initialState.selectedPrinter,
                }

            case CLEAR_PRINTER_DEBUG_OUTPUT:
                return {
                    ...state,
                    debugOutput: []
                }

            case APPEND_PRINTER_DEBUG_OUTPUT:
                return {
                    ...state,
                    debugOutput: [...state.debugOutput, ...action.payload.messages]
                }

            case TOGGLE_DEBUG_OUTPUT:
                return {
                    ...state,
                    isDebugOutputVisible: !state.isDebugOutputVisible
                }

            case DISPLAY_PRINTER_NOTIFICATION:
                return {
                    ...state,
                    notificationTitle: action.payload.title,
                    notificationMessage: action.payload.message,
                    isNotificationVisible: __DEV__,
                    // This functionality is currently only available on a dev build as we still need to implement i18n
                }

            case HIDE_PRINTER_NOTIFICATION:
                return {
                    ...state,
                    isNotificationVisible: false,
                }

            case SET_DEBUG_OUTPUT:
                return {
                    ...state,
                    isDebugOutputVisible: action.payload.value
                }

            default:
                return state
        }
    }

export default reducer
