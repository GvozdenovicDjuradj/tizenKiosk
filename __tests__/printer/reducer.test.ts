import faker from "faker"
import { initialState } from "../../src/printer/reducer"
import {
    appendPrinterDebugOutput,
    clearPrinterDebugOutput,
    displayPrinterNotification,
    findConnectedPrinters,
    findConnectedPrintersFulfilled,
    findConnectedPrintersRejected,
    hidePrinterNotification,
    selectPrinter,
    setDebugOutput,
    setPrinterEnabled, 
    ticketPrinted,
    toggleDebugOutput
} from "../../src/printer/actionCreators"
import reducer from "../../src/printer/reducer"
import { createPrinters } from "./fixtures"
import { Printer } from "@dnlowman/react-native-star-prnt"
import { createCustomerInQueue, createPrinterState } from "../fixtures"
import { printTicket } from "../../src/actions"

describe(`the printer reducer module`, () => {
    it(`should have the following initialState`, () => {
        // Then
        expect(initialState).toMatchSnapshot()
    })

    it(`should react to FIND_CONNECTED_PRINTERS and set isFindingPrinters to true`, () => {
        // Given
        const action = findConnectedPrinters()

        // When
        const state = reducer(initialState, action)

        // Then
        expect(state).toBeDefined()
        expect(state).toHaveProperty("isFindingPrinters", true)
    })

    it(`should react to FIND_CONNECTED_PRINTERS_FULFILLED,
        set isFindingPrinters to false and set connectedPrinters to the new list of printers`, () => {
            // Given
            const printers = createPrinters()
            const action = findConnectedPrintersFulfilled(printers)

            // When
            const state = reducer(initialState, action)

            // Then
            expect(state).toBeDefined()
            expect(state).toHaveProperty("isFindingPrinters", false)
            expect(state).toHaveProperty("connectedPrinters", printers)
        })

    it(`should react to FIND_CONNECTED_PRINTERS_REJECTED and set isFindingPrinters to false`, () => {
        // Given
        const action = findConnectedPrintersRejected()

        // When
        const state = reducer(initialState, action)

        // Then
        expect(state).toBeDefined()
        expect(state).toHaveProperty("isFindingPrinters", false)
    })

    it(`should react to SELECT_PRINTER and set selectedPrinter to the new value`, () => {
        // Given
        const printer: Printer = {
            portName: faker.random.word(),
            modelName: faker.random.word()
        }
        const action = selectPrinter(printer)

        // When
        const state = reducer(initialState, action)

        // Then
        expect(state).toBeDefined()
        expect(state).toHaveProperty("selectedPrinter", printer)
    })

    it(`should react to SET_PRINTER_ENABLED and set isPrinterEnabled and reset selected printer`, () => {
        // Given
        const enabled = faker.random.boolean()
        const action = setPrinterEnabled(enabled)

        // When
        const result = reducer(undefined, action)

        // Then
        expect(result).toBeDefined()
        expect(result).toHaveProperty("isPrinterEnabled", enabled)
        expect(result).toHaveProperty("selectedPrinter", initialState.selectedPrinter)
    })

    it(`should react to CLEAR_PRINTER_DEBUG_OUTPUT and update the state`, () => {
        // Given
        const state = createPrinterState({
            debugOutput: ["Connected", "Fetching template information...", "Failed..."]
        })
        const action = clearPrinterDebugOutput()

        // When
        const result = reducer(state, action)

        // Then
        expect(result).toBeDefined()
        expect(result.debugOutput).toHaveLength(0)
    })

    it(`should react to APPEND_PRINTER_DEBUG_OUTPUT and update the state`, () => {
        // Given
        const message = "Connected"
        const state = createPrinterState({
            debugOutput: []
        })
        const action = appendPrinterDebugOutput(message)

        // When
        const result = reducer(state, action)

        // Then
        expect(result).toBeDefined()
        expect(result.debugOutput).toHaveLength(1)
        expect(result.debugOutput[0]).toContain(message)
    })

    it(`should react to TOGGLE_DEBUG_OUTPUT and update the state`, () => {
        // Given
        const action = toggleDebugOutput()
        const state = createPrinterState()

        // When
        const result = reducer(state, action)

        // Then
        expect(result).toBeDefined()
        expect(result).toHaveProperty("isDebugOutputVisible", !state.isDebugOutputVisible)
    })

    it(`should react to DISPLAY_PRINTER_NOTIFICATION and update the state`, () => {
        // Given
        const title = faker.random.word()
        const message = faker.random.word()
        const action = displayPrinterNotification(title, message)

        // When
        const result = reducer(undefined, action)

        // Then
        expect(result).toBeDefined()
        expect(result).toHaveProperty("notificationTitle", title)
        expect(result).toHaveProperty("notificationMessage", message)
        expect(result).toHaveProperty("isNotificationVisible", true)
    })

    it(`should react to HIDE_PRINTER_NOTIFICATION and update the state`, () => {
        // Given
        const action = hidePrinterNotification()

        // When
        const result = reducer(undefined, action)

        // Then
        expect(result).toBeDefined()
        expect(result).toHaveProperty("isNotificationVisible", false)
    })

    it(`should react to TICKET_PRINTED and update the state`, () => {
        // Given
        const state = createPrinterState()
        const action = ticketPrinted()

        // When
        const result = reducer(state, action)

        // Then
        expect(result).toBeDefined()
        expect(result).toHaveProperty("isPendingPrint", false)
        expect(result).toHaveProperty("isNotificationVisible", false)
        expect(result).toHaveProperty("notificationTitle", "")
        expect(result).toHaveProperty("notificationMessage", "")
    })

    it(`should react to SET_DEBUG_OUTPUT and update the state`, () => {
        // Given
        const value = faker.random.boolean()
        const state = createPrinterState()
        const action = setDebugOutput(value)

        // When
        const result = reducer(state, action)

        // Then
        expect(result).toBeDefined()
        expect(result).toHaveProperty("isDebugOutputVisible", value)
    })

    it(`should react to PRINT_TICKET and update the state`, () => {
        // Given
        const customerInQueue = createCustomerInQueue()
        const value = faker.random.boolean()
        const state = createPrinterState()
        const action = printTicket(customerInQueue, value)

        // When
        const result = reducer(state, action)

        // Then
        expect(result).toBeDefined()
        expect(result).toHaveProperty("isCurrentPrintATest", value)
    })
})
