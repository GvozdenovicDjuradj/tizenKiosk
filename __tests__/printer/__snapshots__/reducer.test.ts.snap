// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the printer reducer module should have the following initialState 1`] = `
Object {
  "connectedPrinters": Immutable.List [],
  "debugOutput": Array [],
  "isCurrentPrintATest": false,
  "isDebugOutputVisible": false,
  "isFindingPrinters": false,
  "isNotificationVisible": false,
  "isPendingPrint": false,
  "isPrinterEnabled": false,
  "localCurrentDate": "",
  "localCurrentTime": "",
  "logoFooterUrl": "",
  "logoUrl": "",
  "notificationMessage": "",
  "notificationTitle": "",
  "selectedPrinter": undefined,
  "showAtTopValue": "",
  "ticketMessage": "",
}
`;
